onnxruntime/test/common/random_generator_test.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "gtest/gtest.h"
+#include "test/common/tensor_op_test_utils.h"
+#include<vector>
+
+namespace onnxruntime {
+namespace test {
+
+TEST(RandomTensorGenerator, DiscreteFloat) {
+  RandomValueGenerator random{};
+  const std::vector<int64_t> shape = {2, 3};
+  std::vector<float> data = random.Discrete<float>(shape, {-1.f, 0.f, 1.f});
+
+  ASSERT_EQ(data.size(), 6);
+  for (float value : data) {
+    EXPECT_TRUE(value == -1.f || value == 0.f || value == 1.f);
+  }
+}
+
+TEST(RandomTensorGenerator, DiscreteInt) {
+  RandomValueGenerator random{};
+  const std::vector<int64_t> shape = {2, 3};
+  std::vector<int> data = random.Discrete<int>(shape, {-1, 0, 1});
+
+  ASSERT_EQ(data.size(), 6);
+  for (int value : data) {
+    EXPECT_TRUE(value == -1 || value == 0 || value == 1);
+  }
+}
+
+TEST(RandomTensorGenerator, DiscreteBool) {
+  RandomValueGenerator random{};
+  const std::vector<int64_t> shape = {2, 3};
+  std::vector<bool> data = random.Discrete<bool>(shape, {false, true});
+
+  ASSERT_EQ(data.size(), 6);
+  for (bool value : data) {
+    EXPECT_TRUE(value == false || value == true);
+  }
```

Comment:
this part of the test does not look very meaningful. I am not sure if there is any good way to test random bool generation. I think we can probably skip this.

Reply:
Good catch! This is generated by ChatGPT. lol